lib/matplotlib/backend_bases.py

```diff
@@ -343,12 +344,11 @@ def _iter_collection_raw_paths(self, master_transform, paths,
         if Npaths == 0:
             return
 
-        transform = transforms.IdentityTransform()
-        for i in range(N):
-            path = paths[i % Npaths]
-            if Ntransforms:
-                transform = Affine2D(all_transforms[i % Ntransforms])
-            yield path, transform + master_transform
+        paths = itertools.cycle(paths)
+        trfs = (itertools.cycle(map(Affine2D, all_transforms)) if Ntransforms
+                else itertools.repeat(transforms.IdentityTransform()))
+        for path, trf in itertools.islice(zip(paths, trfs), N):
+            yield path, trf + master_transform
```

Comment:
This is significantly harder to read than what it replaces.  Is there a significant performance advantage?

Reply:
This was mostly changed for consistency with _iter_collection (where I'd argue that having all the cycling behavior in one place definitely improves readability); here I agree the benefit is less clear and can revert that if you prefer.

Reply:
I agree with @jklymak: the original version is more readable (unless you are a functional programming enthusiast).

Reply:
one vote for the change? I find the new version clearer - for me, the modulo indexing based cycling took a couple of minutes to parse first time I saw it.

Reply:
Sure, restored the old version here.

Reply:
Sorry @story645, that came in too late :-)

Reply:
It's probably fine if you are intimately familiar with itertools, but I definitely am not, whereas I have no problem with modulo.

Reply:
itertools is the best, but 🤷‍♀️ I like the cleanup either way

Reply:
I'm not sure it would be measurable here, but I've definitely run into code where `itertools` was >> "doing math" in terms of runtime performance. I only even mention it since we're in the render loop.

Reply:
@dopplershift i can't parse your statement as itertools better or worse

Reply:
Generally, `itertools` better, but not no idea if my runtime argument is meaningful here or not.